### DISCUSSION TITLE
Native Bluesound/NAD player support?

### DISCUSSION THREAD

**Discussion by bluefuzz** [music-assistant/support#2245] [Feature requests and ideas]
What is the chance of native Bluesound/NAD player support, i.e. BluOS? There is a HA BluOS player but that seems to require flow-mode to work properly. I can use the Airplay option but that is limited to 16/44.1. There isn't even a poll option for BluOS which is a bit surprising. I would have thought Bluesound devices are every bit as popular as Sonos etc. ...

**Comment by OzGav** [comment]
It’s on here now but I have never heard of it before. Looking at the HA analytics. The Bluesound integration is #426 with 0.1% of installations using it. HEOS is #122 with 2.6%. If a dev comes along to pick this up that will be great but just being realistic that core effort will likely go to HEOS first.

**Reply by bluefuzz** [reply]
OK, fair enough. Surprising you've never heard of Bluesound or NAD - one of the biggest names in Hifi for decades. Personally I've never heard of HEOS ... ;-)

FWIW BluOS has a documented API: https://bluos.io/downloads/

**Reply by OzGav** [reply]
I’ve heard of NAD just not BlueSound. Don’t worry I use Yamaha which is at #521 with 0.0% 😳

**Reply by bluefuzz** [reply]
Both NAD and Bluesound are subsidiaries of Lenbrook. NAD mostly does the traditional hifi stuff and Bluesound does the 'smart speaker' lifestyle stuff. Both use BluOS as operating systems where appropriate in their streaming appliances. I believe Dali loudspeakers also use BluOS too.

**Comment by SamBouwer** [comment]
As a Bluesound user I would also be happy to see native integration with BluOS. There is an integration available to connect to Bluesound speakers directly, so maybe that logic can be reused? https://www.home-assistant.io/integrations/bluesound/

**Reply by OzGav** [reply]
Bluesound is now available https://music-assistant.io/player-support/bluesound/

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
